### ISSUE TITLE
How to replace Node innerHTML?

### ISSUE THREAD

**Issue by TangMonk** [benibela/internettools#22]
Is it able to replace node innerHTML, like this:

```
for node in process(everydayHtmlString, '//table[@id="zhuye" or @id="fuye"]') do
  begin
    if node.toNode.getAttribute('id') = 'zhuye' then
    begin
         node.toNode.outerHTML := aStringContainHTML;
    end;
  end;
```

**Comment by benibela** [comment]
No, once the document is created, it cannot be changed anymore.

The you can use the transform function in xquery to create a new document (needs xquery_json in uses)

    process(everydayHtmlString, ' x:transform(., function($node) { if ($node/self::table[@id="zhuye"]) then <table>aStringContainHTML</table> else $node }  )  ');

**Comment by TangMonk** [comment]
@benibela how about add a `setInnerHTML`  method for Node?

**Comment by TangMonk** [comment]
I am used Winform in C#, there is a libaray called [Html Agility Pack](https://html-agility-pack.net/), It is really convient and easy to use for parse a web page

**Comment by benibela** [comment]
I do not want people to believe they can change individual nodes. Then they might try multithreading, change different nodes in different threads and the program fails because you can only modify the entire document.

Every node has an index, first node might be 10,  second node 20, third node 33, ...

If you add a node, it needs to update all the node indices afterwards.
